app/api/suggestions/blankSuggestions.ts

```diff
@@ -75,25 +120,58 @@ export const createBlankSuggestionsForPartialExtractor = async (
   const templatesPromises = selectedTemplates
     .filter(template => extractorTemplates.has(template.toString()))
     .map(async template => {
-      const entitiesSharedIds = await fetchEntitiesSharedIds(template, defaultLanguage, batchSize);
+      const suggestionsToSave: IXSuggestionType[] = [];
 
-      const fetchedFiles = await files.get(
-        { entity: { $in: entitiesSharedIds }, type: 'document' },
-        '_id entity language extractedMetadata'
-      );
+      if (extractor.source.pdf) {
+        const entityDataForFiles = await fetchEntitiesData(template, defaultLanguage, batchSize);
 
-      const suggestionsToSave: IXSuggestionType[] = fetchedFiles
-        .filter(file => file.entity)
-        .map(file =>
-          getBlankSuggestion(file, extractor, template, exampleProperty.type, defaultLanguage)
+        const fetchedFiles = await files.get(
+          { entity: { $in: entityDataForFiles.map(entity => entity.sharedId) }, type: 'document' },
+          '_id entity language extractedMetadata'
         );
+
+        suggestionsToSave.push(
+          ...fetchedFiles
+            .filter(file => file.entity)
+            .map(file =>
+              getBlankSuggestionForPdf({
+                file,
+                extractorId: extractor._id,
+                template,
+                propertyName: extractor.property,
+                propertyType: exampleProperty.type,
+                defaultLanguage,
+              })
+            )
+        );
+      } else if (tenants.current().featureFlags?.ixExtraSources) {
+        const entityData = await fetchEntitiesData(template, undefined, batchSize);
+
+        suggestionsToSave.push(
+          ...entityData.map(entity =>
+            getBlankSuggestionForProperty({
+              entityId: entity.sharedId,
+              extractorId: extractor._id,
+              template,
+              propertyName: extractor.property,
+              propertyType: exampleProperty.type,
```

Comment:
This is not from you, but the naming "exampleProperty" is not really ideal, right?  Should it be `sampleProperty`?